src/main/java/org/jabref/gui/externalfiles/AutoLinkFilesAction.java

```diff
@@ -34,47 +33,62 @@
+        this.dialogService = dialogService;
         this.preferences = preferences;
         this.stateManager = stateManager;
         this.undoManager = undoManager;
         this.taskExecutor = taskExecutor;
 
         this.executable.bind(needsDatabase(this.stateManager).and(needsEntriesSelected(stateManager)));
         this.statusMessage.bind(BindingsHelper.ifThenElse(executable, "", Localization.lang("This operation requires one or more entries to be selected.")));
-
     }
 
     @Override
     public void execute() {
-        BibDatabaseContext database = stateManager.getActiveDatabase().orElseThrow(() -> new NullPointerException("Database null"));
-        List<BibEntry> entries = stateManager.getSelectedEntries();
-
-        final NamedCompound nc = new NamedCompound(Localization.lang("Automatically set file links"));
-        AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(
+        final BibDatabaseContext database = stateManager.getActiveDatabase().orElseThrow(() -> new NullPointerException("Database null"));
+        final List<BibEntry> entries = stateManager.getSelectedEntries();
+        final AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(
                 database,
                 preferences.getFilePreferences(),
                 preferences.getAutoLinkPreferences(),
                 ExternalFileTypes.getInstance());
-        Task<List<BibEntry>> linkFilesTask = new Task<>() {
+        final NamedCompound nc = new NamedCompound(Localization.lang("Automatically set file links"));
+
+        if (entries.stream().mapToInt(entry -> entry.getFiles().size()).sum() == 0) {
+            dialogService.notify(Localization.lang("The selected entries do not have any file links attached."));
+            return;
+        }
 
+        Task<AutoSetFileLinksUtil.LinkFilesResult> linkFilesTask = new Task<>() {
             @Override
-            protected List<BibEntry> call() {
+            protected AutoSetFileLinksUtil.LinkFilesResult call() {
                 return util.linkAssociatedFiles(entries, nc);
             }
 
             @Override
             protected void succeeded() {
-                if (!getValue().isEmpty()) {
-                    if (nc.hasEdits()) {
-                        nc.end();
-                        undoManager.addEdit(nc);
-                    }
-                    dialogService.notify(Localization.lang("Finished automatically setting external links."));
-                } else {
-                    dialogService.notify(Localization.lang("Finished automatically setting external links.") + " " + Localization.lang("No files found."));
+                AutoSetFileLinksUtil.LinkFilesResult result = getValue();
+
+                if (!result.getFileExceptions().isEmpty()) {
+                    dialogService.notify(Localization.lang("Problem finding files. See error log for details."));
```

Comment:
I think this should be a normal dialog. I would reserve `notify` for "everything went well"-messages.